clients/roscpp/include/ros/timer_manager.h

```diff
@@ -560,10 +574,21 @@ void TimerManager<T, D, E>::threadFunc()
       }
       else
       {
-        // On system time we can simply sleep for the rest of the wait time, since anything else requiring processing will
-        // signal the condition variable
-        int32_t remaining_time = std::max((int32_t)((sleep_end - current).toSec() * 1000.0f), 1);
-        timers_cond_.timed_wait(lock, boost::posix_time::milliseconds(remaining_time));
+        // we have to distinguish between SteadyTime and WallTime here,
+        // because boost timed_wait with duration just adds the duration to current system time
+        // this however requires boost 1.61, see: https://svn.boost.org/trac/boost/ticket/6377
+        if (typeid(T) == typeid(SteadyTime))
```

Comment:
If you want to check the type `T` and only accept exactly `SteadyTime`. That could be done with some template meta programming since the function is already templated on `T`.

Or the `SteadyTime` class could have a method which return `true` for that class (and `false` for the other variations). It might not be possible to introduce that in an ABI compatible fashion which would prevent backporting the change.